tensorvis/tensorvisio/src/processors/vtkdatasettotensorfield3d.cpp

```diff
@@ -0,0 +1,155 @@
+
+VTKDataSetToTensorField3D::VTKDataSetToTensorField3D()
+    : Processor()
+    , dataSetInport_("dataSetInport")
+    , tensorField3DOutport_("tensorField3DOutport")
+    , normalizeExtents_("normalizeExtents", "Normalize extents", false)
+    , arrays_("arrays", "arrays")
+    , offset_{0}
+    , previouslySelectedArrayName_{} {
+    addPort(dataSetInport_);
+    addPort(tensorField3DOutport_);
+
+    addProperty(normalizeExtents_);
+
+    addProperty(arrays_);
+
+    dataSetInport_.onChange([this]() { initializeResources(); });
+}
+
+void VTKDataSetToTensorField3D::initializeResources() {
+    if (!dataSetInport_.hasData()) return;
+
+    const auto& dataSet = *dataSetInport_.getData();
+    auto pointData = dataSet->GetPointData();
+
+    auto stringReplace = [](std::string str, const std::string& from,
+                            const std::string& to) -> const std::string {
+        size_t start_pos{0};
+        while ((start_pos = str.find(from, start_pos)) != std::string::npos) {
+            str.replace(start_pos, from.length(), to);
+            start_pos += to.length();
+        }
+        return str;
+    };
+
+    arrays_.clearOptions();
+
+    for (int i{0}; i < pointData->GetNumberOfArrays(); ++i) {
+        auto array = pointData->GetArray(i);
+
+        if (array->GetNumberOfComponents() != 9) continue;
+
+        arrays_.addOption(stringReplace(stringReplace(array->GetName(), ".", ""), " ", ""),
+                          {array->GetName()}, {array->GetName()});
+    }
+
+    // If the data set contains an array with the same name as the previous data set, we set the
+    // selected option to that array. This is mostly for serialization purposes.
+    for (const auto& option : arrays_.getOptions()) {
+        if (option.id_ == previouslySelectedArrayName_) {
+            arrays_.setSelectedValue(option.value_);
+            return;
+        }
+    }
+    arrays_.setSelectedIndex(0);
+    previouslySelectedArrayName_ = arrays_.getOptionDisplayName(0);
+}
+
+void VTKDataSetToTensorField3D::process() {
+    if (!dataSetInport_.hasData()) return;
+    dispatchPool([this]() {
+        const auto& dataSet = *dataSetInport_.getData();
+
+        auto dataArray = dataSet->GetPointData()->GetArray(arrays_.get().c_str());
+
+        const auto dimensions = dataSet.getDimensions();
+
+        LogProcessorInfo("Attempting to generate tensor field from array \""
+                         << std::string{dataArray->GetName()} << "\"");
+
+        const auto bounds = dataSet->GetBounds();
+        auto extent = tensorutil::vtk_ExtentFromBounds(bounds);
+        const auto offset = tensorutil::vtk_OffsetFromBounds(bounds);
+
+        if (normalizeExtents_.get()) {
+            extent /= std::max(std::max(extent.x, extent.y), extent.z);
+        }
+
+        std::shared_ptr<TensorField3D> tensorField;
+        if (dataArray->GetDataType() == VTK_DOUBLE) {
```

Comment:
I am guessing that this will be replaced at some point by a dispatch?

Reply:
If somebody cares enough. It covers my use cases for the foreseeable future.
Using dispatching here would mean that the tensor field class needs to be templated and float/double is really all we care about.